### DISCUSSION TITLE
A plugin to disable the annoying layout switch window

### DISCUSSION THREAD

**Discussion by Evengard** [valinet/ExplorerPatcher#1455] [Ideas]
First of all because I got lost whenever I should create a Discussion or an Issue (the strikethrough is really making it all worse), and also because I got fed up with this horrible Windows behaviour, I made a plugin for fixing that problem myself.

The language switch window which appears when you press your language switch keybind (aka Alt+Shift or Ctrl+Shift) was so annoying, and even more it somehow bugged the actual language switching, preventing the actual switch until that window appeared!

So I used the ep_extra.dll infrastructure and made this: https://github.com/Evengard/ep_extra_evengard.win11disableinputswitchwindow

This is supposed to be compiled as a submodule of the main ExplorerPatcher repo.

Uses Detours for the hooking and DbgHelp+PDBDownloader to detect what exactly should be overriden.

If the author wants to integrate that in the main codebase, I don't mind that at all (and would actually be pretty happy about that).

If users want to try this out as a user, the [Releases page](https://github.com/Evengard/ep_extra_evengard.win11disableinputswitchwindow/releases) have everything you'd need.

**Comment by pyrates999** [comment]
Well done!
